src/libraries/System.Private.CoreLib/src/System/Globalization/Normalization.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Text;
+
+namespace System.Globalization
+{
+    internal static partial class Normalization
+    {
+        internal static bool IsNormalized(string strInput, NormalizationForm normalizationForm)
+        {
```

Comment:
Factor our the `if (GlobalizationMode.Invariant)` call into this method as well?